### PR TITLE
[Feature] Create/Edit experience page

### LINKED ISSUE #6935: 📃 Update the "Create or edit experience" page
image

<details><summary>📸 Full page screenshot (create)</summary>

</details>

<details><summary>📸 Full page screenshot (edit)</summary>

</details>

# ✨ Details ([Figma file](https://www.figma.com/file/piRkRtWAHheoEX5Jq8d9u7/Candidates---Create-or-edit-experience?type=design&node-id=64-107861))

This ticket aims to update the functionality of adding and editing a résumé experience to match the new application process, alongside a few new enhancements.

**Please note:** Where relevant, features that are described here and that should NOT translate to the application experience have been marked with a ⚠️ emoji.

# 🗃️ General requirements

```[tasklist]
## 🌎 Page-wide updates
- [ ] Ensure all copy is updated to match
- [ ] Ensure font sizes are all updated to match
- [ ] Ensure icons are updated and match sizing and color
``` 

# 🗃️ Page section requirements

```[tasklist]
## 📚 Instructions
- [ ] Ensure all copy is updated to match
- [ ] Ensure icons are updated and match sizing and color
```

<details><summary>📸 Instructions - Create - Collapsed</summary>

</details> 

<details><summary>📸 Instructions - Create - Expanded</summary>

</details> 

```[tasklist]
## 📚 Experience type selection
- [ ] Ensure all copy is updated to match
``` 

<details><summary>📸 Select an experience - Create - Expanded</summary>

</details> 

```[tasklist]
## 📚 Experience details
- [ ] Ensure all copy is updated to match
- [ ] Prevent the user from seeing a form until a type is selected
- [ ] Update all experience form layouts to match
``` 

<details><summary>📸 Provide a few details - Locked state</summary>

</details> 

<details><summary>📸 Provide a few details - Work experience form</summary>

</details> 

<details><summary>📸 Provide a few details - Education experience form</summary>

</details> 

<details><summary>📸 Provide a few details - Community experience form</summary>

</details> 

<details><summary>📸 Provide a few details - Personal experience form</summary>

</details> 

<details><summary>📸 Provide a few details - Award experience form</summary>

</details> 

```[tasklist]
## 📚 Additional info
- [ ] Ensure all copy is updated to match
- [ ] Prevent the user from seeing a form until a type is selected
``` 

<details><summary>📸 Additional details - Locked</summary>

</details> 

<details><summary>📸 Additional details - Unlocked</summary>

</details> 

```[tasklist]
## 📚 Link featured skills ⚠️
- [ ] Prevent the user from seeing a form until a type is selected
- [ ] Allow for skills linked via an application to be edited here
- [ ] Build a null state for when users have no skills linked yet
- [ ] Allow users to add new skills to the experience by implementing the skill browser dialog
- [ ] Allow users to delete skills from the experience
``` 

**Please note:**
- Featured skills use the same linking functionality designed for the application when applying a skill justification

<details><summary>📸 Featured skills - Locked - Collapsed</summary>

</details> 

<details><summary>📸 Featured skills - Locked - Expanded</summary>

</details> 

<details><summary>📸 Featured skills - Unlocked - No skills</summary>

</details> 

<details><summary>📸 Featured skills - Unlocked - With skills</summary>
 
</details> 

### The skill browser dialog

When clicking "Add a new item", the skill browser dialog should be triggered and should include:
- Skill category filter
- Skill family filter
- Skill combobox
- Skill definition
- Expansion describing what to do if the user can't find a skill

See issue #5053 for more details.

<details><summary>📸 Skill browser dialog</summary>

</details>

# 🗃️ State and other requirements

```[tasklist]
## 🔄️ The differences between creating and editing
- [ ] Update the page title, subtitle, and breadcrumbs
- [ ] Update the navigation sidebar
- [ ] Update the main section title and instruction content
- [ ] Remove the ability (and section) to select the type of experience
- [ ] Update the page actions to remove "Save and create another" and add "Delete this experience"
```

Comparison screenshots can be found below for visual and text differences.

<details><summary>📸 Creating vs. editing - Banner</summary>
 
</details> 

<details><summary>📸 Creating vs. editing - Sidebar</summary>
 
</details>

<details><summary>📸 Creating vs. editing - Instructions</summary>
 
</details> 

<details><summary>📸 Creating vs. editing - Hidden type selection</summary>
 
</details> 

<details><summary>📸 Creating vs. editing - Page actions</summary>
 
</details>

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #6935

## 👋 Introduction

Aims to update the functionality of adding and editing a résumé experience to match the new application process, alongside a few new enhancements.

- Refactors the ExperienceFormPage component, adding both create and edit functionality 
- Adds new [SkillDialog](https://github.com/GCTC-NTGC/gc-digital-talent/issues/5053) component (minor changes made which might need to be updated)
- Updates routing on profile->resume-and-requirements->add/edit. Some paths needed to be updated too, removing the experienceType from the path, since the user can change the experienceType on the form.

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

1. Login as admin/applicant
2. Go to Profile and Applications in nav, then Resume and Recruitments page.
3. Try adding or editing an experience
4. Ensure all requirements are met in issue.

### CODE REVIEW

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+              defaultMessage: "Edit an experience",
+              id: "RzXSJ9",
+              description:
+                "Button to open modal to add a new experience to the profile",
+            })}
+          </Heading>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage:
+                "This form allows you to edit a specific experience, its details, and linked skills. Don’t forget, work experiences should focus on describing  your part-time, full-time, self-employment, fellowship, non-profit, or internship experiences.",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:52]:
```suggestion
                "This form allows you to edit a specific experience, its details, and linked skills. Don’t forget, work experiences should focus on describing your part-time, full-time, self-employment, fellowship, non-profit, or internship experiences.",
```

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+              defaultMessage: "Add a new experience",
+              id: "bOAF9o",
+              description:
+                "Button to open modal to add a new experience to the profile",
+            })}
+          </Heading>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage:
+                "This form allows you to add a new experience to your résumé. Get started by selecting the type of experience you’d like to add. If you need more information about what a certain type can include, expand the information below to see examples.",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:79]:
```suggestion
                "This form allows you to add a new experience to your career timeline. Get started by selecting the type of experience you’d like to add. If you need more information about what a certain type can include, expand the information below to see examples.",
```

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+              id: "F5KhNJ",
+              description:
+                "Instructions on how to add an experience to your résumé",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:82]:
```suggestion
                "Instructions on how to add an experience to your career timeline",
```

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            value={accordionState}
+            onValueChange={(value: AccordionStates) => setAccordionState(value)}
+            collapsible
+          >
+            <Accordion.Item value="learn-more">
+              <StandardAccordionHeader headingAs="h3">
+                {accordionState === "learn-more"
+                  ? intl.formatMessage({
+                      defaultMessage:
+                        "Hide more information about the types of experience you can add to your résumé",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:97]:
```suggestion
                        "Hide more information about the types of experience you can add to your career timeline",
```

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                      description:
+                        "Button text to open section describing experience types",
+                    })}
+                {}
+              </StandardAccordionHeader>
+              <Accordion.Content>
+                <p data-h2-margin-top="base(x1)">
+                  {intl.formatMessage({
+                    defaultMessage:
+                      "We all have a variety of accomplishments and experiences that shape both our careers and skills. Please only share what you would be comfortable sharing with a coworker. On this platform, you can add the following to your résumé:",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:115]:
```suggestion
                      "We all have a variety of accomplishments and experiences that shape both our careers and skills. Please only share what you would be comfortable sharing with a coworker. On this platform, you can add the following to your career timeline:",
```

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                        "Description for personal experience section",
+                    })}
+                  </DefinitionList.Item>
+                  <DefinitionList.Item
+                    Icon={StarIcon}
+                    title={intl.formatMessage(experienceTypeTitles.award)}
+                  >
+                    {intl.formatMessage({
+                      defaultMessage:
+                        "Share specific awards and recognition you've received for going above and beyond.",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:175]:
```suggestion
                        "Share specific awards and recognition that you've received for going above and beyond.",
```

File: apps/web/src/components/ExperienceFormFields/SelectExperience.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  const derivedType = type ?? experienceType;
+
+  return (
+    <section data-h2-margin="base(0, 0, x2, 0)">
+      <Heading level="h3" size="h5" data-h2-margin-bottom="base(x.5)">
+        {intl.formatMessage({
+          defaultMessage: "Select a type of experience",
+          id: "jw6Umr",
+          description:
+            "Heading for the experience type section fo the experience form",
```
Comment (mnigh) [inline apps/web/src/components/ExperienceFormFields/SelectExperience.tsx:29]:
```suggestion
            "Heading for the experience type section for the experience form",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx [hunk 8]
```diff
@@ -138,222 +145,244 @@ export const ExperienceForm = ({
...
-          id: "Ytlyzb",
-          description: "Page title for the work experience profile form",
-        });
-      default:
-        return "";
-    }
-  };
+  const pageTitle: string = experience
+    ? intl.formatMessage({
+        defaultMessage: "Edit a résumé experience",
```
Comment (mnigh) [inline apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx:154]:
```suggestion
        defaultMessage: "Edit a career timeline experience",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx [hunk 9]
```diff
@@ -138,222 +145,244 @@ export const ExperienceForm = ({
...
+        id: "UISX7F",
+        description: "Display text for edit experience form in breadcrumbs",
+      })
+    : intl.formatMessage({
+        defaultMessage: "Add an experience to your résumé",
```
Comment (mnigh) [inline apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx:159]:
```suggestion
        defaultMessage: "Add an experience to your career timeline",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx [hunk 10]
```diff
@@ -138,222 +145,244 @@ export const ExperienceForm = ({
...
+        id: "s9hqAQ",
+        description: "Display text for add experience form in breadcrumbs",
+      });
+
+  const pageSubtitle: string = experience
+    ? intl.formatMessage({
+        defaultMessage: "Update or delete an experience on your résumé",
```
Comment (mnigh) [inline apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx:166]:
```suggestion
        defaultMessage: "Update or delete an experience on your career timeline",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx [hunk 11]
```diff
@@ -138,222 +145,244 @@ export const ExperienceForm = ({
...
+                  data-h2-flex-wrap="base(wrap)"
+                  data-h2-flex-direction="base(column) l-tablet(row)"
+                  data-h2-align-items="base(flex-start) l-tablet(center)"
+                >
+                  <Button type="submit">
                     {intl.formatMessage({
-                      defaultMessage: "Cancel",
-                      id: "KnE2Rk",
-                      description: "Cancel confirmation",
+                      defaultMessage: "Save and return to my résumé",
```
Comment (mnigh) [inline apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx:273]:
```suggestion
                      defaultMessage: "Save and return to my career timeline",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx [hunk 12]
```diff
@@ -138,222 +145,244 @@ export const ExperienceForm = ({
...
-                    onClick={deleteExperience}
+                    value="return"
+                    {...actionProps}
+                    onClick={() => setValue("action", "return")}
                   >
                     {intl.formatMessage({
-                      defaultMessage: "Delete",
-                      id: "sBksyQ",
-                      description: "Delete confirmation",
+                      defaultMessage: "Save and return to my résumé",
```
Comment (mnigh) [inline apps/web/src/pages/Profile/ExperienceFormPage/ExperienceFormPage.tsx:354]:
```suggestion
                      defaultMessage: "Save and return to my career timeline",
```

**Review by mnigh — DISMISSED** [review]
My review is mostly a fairly light copy edit. A lot of the suggestions are based on the change from résumé to career timeline that was was part of https://github.com/GCTC-NTGC/gc-digital-talent/pull/7305.

File: apps/web/src/pages/Profile/ExperienceFormPage/components/ExperienceSkills.tsx [hunk 13]
```diff
@@ -80,139 +65,158 @@ const ExperienceSkills = ({ skills, pool }: ExperienceSkillsProps) => {
...
+          ) : (
+            <Well>
+              <p
+                data-h2-margin-bottom="base(x1)"
+                data-h2-text-align="base(center)"
+                data-h2-font-weight="base(bold)"
+              >
+                {intl.formatMessage({
+                  defaultMessage:
+                    "You haven’t featured any skills on this experience yet.",
```
Comment (esizer) [inline apps/web/src/pages/Profile/ExperienceFormPage/components/ExperienceSkills.tsx:198]: Found a smart apostrophe😅 

```suggestion
                    "You haven't featured any skills on this experience yet.",
```

File: apps/web/src/pages/Profile/ExperienceFormPage/components/ExperienceSkills.tsx [hunk 14]
```diff
@@ -80,139 +65,158 @@ const ExperienceSkills = ({ skills, pool }: ExperienceSkillsProps) => {
...
+                  description:
+                    "Primary message to user when no skills have been attached to experience.",
+                })}
+              </p>
+              <p data-h2-text-align="base(center)">
                 {intl.formatMessage({
-                  defaultMessage: "Nice-to-have skills",
-                  id: "xQ2Vfi",
+                  defaultMessage:
+                    "You can use the “Add a skill” button provided to feature skills here.",
```
Comment (esizer) [inline apps/web/src/pages/Profile/ExperienceFormPage/components/ExperienceSkills.tsx:207]: Found some smart quotes 😅 
```suggestion
                    "You can use the "Add a skill" button provided to feature skills here.",
```

**Review by esizer — CHANGES_REQUESTED** [review]
Nice, this is looking really great so far! 😎  I found some minor things for you to take a look at.

File: apps/web/src/components/SkillsInDetail/SkillsInDetail.tsx [hunk 15]
```diff
@@ -1,166 +1,91 @@
...
+              <p data-h2-font-weight="base(700)">
+                {index + 1}. {name[locale]}
+              </p>
+              <Button
+                color="error"
+                mode="inline"
+                data-h2-text-decoration="base(underline)"
+                onClick={() => {
+                  onDelete(skillId);
+                }}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SkillsInDetail/SkillsInDetail.tsx:46]: Looks good, just need to pass the name into `intl.formatMessage`

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+      ) : (
+        <>
+          <Heading data-h2-margin-top="base(0)" data-h2-display="base(flex)">
+            <PlusCircleIcon
+              data-h2-color="base(primary)"
+              data-h2-width="base(x1.5)"
+              data-h2-display="base(inline-block)"
+              data-h2-vertical-align="base(middle)"
+              data-h2-margin="base(0, x.25, 0, 0)"
+            />
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx]: Looks great, just need to add the color now `color="primary"`

File: apps/web/src/components/SkillsInDetail/SkillsInDetail.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/SkillsInDetail/SkillsInDetail.tsx:46]: Nice catch

File: apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+            })}
+          </p>
+        </>
+      ) : (
+        <>
+          <Heading
+            Icon={PlusCircleIcon}
+            data-h2-margin-top="base(0)"
+            data-h2-display="base(flex)"
+          >
```
Comment (esizer) [inline apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx:63]: Sorry, I did not realize this used two different `Heading`'s here. We will need `color="primary"` on this one as well.

```suggestion
          <Heading
            Icon={PlusCircleIcon}
            color="primary"
            data-h2-margin-top="base(0)"
            data-h2-display="base(flex)"
          >
```

Comment (yonikid15) [reply apps/web/src/components/ExperienceFormFields/ExperienceHeading.tsx]: Nice catch

**Review by esizer — APPROVED** [review]
Awesome! ⭐

**Pull request merged**